lib/bing_translator.rb

```diff
@@ -16,18 +16,21 @@ class BingTranslator
   NAMESPACE_URI = 'http://api.microsofttranslator.com/V2'
   ACCESS_TOKEN_URI = 'https://datamarket.accesscontrol.windows.net/v2/OAuth2-13'
   DATASETS_URI = "https://api.datamarket.azure.com/Services/My/Datasets?$format=json"
+  COGNITIVE_ACCESS_TOKEN_URI = 'https://api.cognitive.microsoft.com/sts/v1.0/issueToken'
 
   class Exception < StandardError; end
   class AuthenticationException < StandardError; end
 
-  def initialize(client_id, client_secret, skip_ssl_verify = false, account_key = nil)
+  def initialize(client_id: nil, client_secret: nil, skip_ssl_verify: false, account_key: nil, subscription_key: nil)
```

Comment:
@relrod and I discussed this and we're both not sure if 1.9.3 support is important, but it seems like it would be reasonable to avoid 2.x features for now just in case. Thankfully, with `client_id`/`client_secret` deleted, it won't be very many arguments anyway.